File: frontend/src/api/operationLog.ts
import { http } from "@/utils/http";

/**
 * 统一响应格式
 */
type Result<T = any> = {
  code: number;
  message: string;
  data: T;
};

/**
 * 操作日志信息
 */
export type OperationLogInfo = {
  id: number;
  userId?: number;
  username?: string;
  module?: string;
  operation?: string;
  method?: string;
  url?: string;
  params?: string;
  result?: string;
  status: number;
  ip?: string;
  duration?: number;
  operationTime: string;
};

/**
 * 操作日志列表查询请求参数
 */
export type OperationLogListRequest = {
  current?: number;
  size?: number;
  username?: string;
  module?: string;
  operation?: string;
  status?: number;
  startTime?: string;
  endTime?: string;
};

/**
 * 获取操作日志列表
 */
export const getOperationLogList = (params: OperationLogListRequest) => {
  return http.request<Result<{
    records: OperationLogInfo[];
    total: number;
    current: number;
    size: number;
    pages: number;
  }>>("get", "/api/operation-logs", { params });
};

File: backend/aihub-admin/src/main/java/com/aihub/admin/entity/OperationLog.java
package com.aihub.admin.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 操作日志实体类
 */
@Data
@TableName("operation_log")
public class OperationLog {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    @TableField("user_id")
    private Long userId;
    
    private String username;
    
    private String module;
    
    private String operation;
    
    private String method;
    
    private String url;
    
    private String params;
    
    private String result;
    
    private Integer status;
    
    private String ip;
    
    private Integer duration;
    
    @TableField("operation_time")
    private LocalDateTime operationTime;
}
